cmd/kube-controller-manager/controller-manager.go

```diff
@@ -36,6 +37,13 @@ import (
 )
 
 func main() {
+	if err := runControllerManagerCmd(); err != nil {
```

Comment:
Exceptions could cause this to be skipped. If your going to make this sort of fix this should be a defer.

Reply:
Sorry, but I don't understand what you said.
In case of exceptions (i.e. panic), defer will not be executed. 
So, if I do this logging + os.Exit in defer, it will also be skipped when panic.

Reply:
Ah, i misunderstood. defer func will  not be skipped when panic.
Did you mean we should also print panic message to log with using defer + recover?

Reply:
You can have main start with something like 
```
var err error = nil 
defer func(){
  if err != nil {
      fmt.Fprintf(os.Stderr, "%v\n", err)
      os.Exit(1)
    }
  }()
```
Since the exception did not occur inside the function it should still be invoked.

Reply:
(Sorry to bother you again. )
Thanks for the example code, but, I'm still not sure what your code is meant to do, sorry. 
I still think we can print log and exit whenever runControllerManagerCmd returns error. 

If an exception is thrown on runControllerManagerCmd, we cannot log with L4 of your example code either way. This is because `err` will always be nil with the current code when runControllerManagerCmd() throw panic.
We can catch message from panic if we use recover, but I think we don't need to do that because Go always write panic message to strerr.

And you also said, 
> This code will still have issues is something does an Exit.
https://github.com/kubernetes/kubernetes/pull/104751#discussion_r704934324

yes, so I changed the code in runControllerManagerCmd to return error instead of os.Exit. We have to keep os.Exit unused. And this problem cannot be solved by using your example code. (because defer func will not be called if os.Exit.)

I'm sorry if I'm missing the point😢, and could you elaborate a bit more on what you're thinking with this code, please?